[YW] Added initContainer to copy the prometheus configuration

baba230896 opened the pull request:
### Summary

- Added `initContainer` in the YW helm chart to copy the Prometheus configuration.

### Test Plan

- Deployed the YW & spin up the YB universe on GCP. Updated the Prometheus configuration & is reflected in the Prometheus.

fixes: https://github.com/yugabyte/yugabyte-db/issues/7515

iSignal commented on .github/workflows/update-version.yaml:
```diff
@@ -0,0 +1,56 @@
+        git config user.email 'yugabyte-ci@users.noreply.github.com'
+    - name: "Extract version number from tag"
+      id: extract-version
+      run: |
```
minor: perhaps at a later point, we could create a common workflow for all these version update workflows that does the first couple of steps that all these workflow-update tasks need to do

bhavin192 commented:
Closing this in favour of https://github.com/yugabyte/charts/pull/2

jvigil-yugabyte commented on stable/yugaware/templates/service.yaml:
```diff
@@ -99,6 +99,22 @@ spec:
             secretName: {{ .Release.Name }}-yugaware-tls-cert
         {{- end }}
+      initContainers:
+        - image: busybox:1.33.0
```
In the Yugabyte helm chart, we require busybox:1.31 (https://github.com/yugabyte/charts/blob/master/stable/yugabyte/templates/service.yaml#L392). Requiring two separate busybox images is a bit strange, especially for airgapped installations which will require the administrator to provide both. Should we use the same image in both places? Also, would we want to make this image + tag configurable via the values.yaml file like our other images? CC: @iSignal

jvigil-yugabyte replied:
Also, just noticed https://github.com/yugabyte/charts/pull/54, which removes the other busybox image and replaces it with the yugabyte image. Maybe we would we want to use the yugaware image here, then?

iSignal replied:
Agree that reusing an existing container would make it easier to deploy in offline environments where specific approved containers from an internal repo can be used.

iSignal reviewed (changes requested):
use a diff base image

baba230896 committed:
[Platform charts] Added initContainer to copy the prometheus conf

- Set the initContainer image to YW image as per the review comment.
- Resolved the merge conflicts due to restructuring of platform charts.

baba230896 commented:
@iSignal 
Could you please review it once whenever you've time? This is a very straightforward change.

skorobogatydmitry commented on stable/yugaware/templates/statefulset.yaml:
```diff
@@ -64,6 +64,22 @@ spec:
           secret:
             secretName: {{ .Release.Name }}-yugaware-tls-cert
         {{- end }}
+      initContainers:
```
Suggestion: https://github.com/elastic/helm-charts/blob/master/logstash/templates/statefulset.yaml#L149

This approach is more flexible.

iSignal replied:
is the suggestion that we customize the init containers through values? For our current purpose, the init container here seems reasonable.

bhavin192 replied:
Yes, I agree with Sanketh. The current initContainer is something which is always going to be there in every chart installation.

skorobogatydmitry replied:
Yes, suggestion was to have `initContainers` in Values.
This allows to tune pods without re-building image and overall helpful.
And you're right - it's not required for this particular patch.

skorobogatydmitry commented on stable/yugaware/templates/statefulset.yaml:
```diff
@@ -64,6 +64,22 @@ spec:
+          name: prometheus-configuration
+          command: [ "/bin/sh", "-c"]
+          args:
+            - "cp /default_prometheus_config/prometheus.yml /prometheus_configs/prometheus.yml"
```
Why not use `cp` directly ?

baba230896 replied:
Yes, we can use `cp` directly. Will update it.

skorobogatydmitry commented on stable/yugaware/templates/statefulset.yaml:
```diff
@@ -130,7 +146,7 @@ spec:
           {{- end }}
           command: [ "/bin/bash", "-c"]
           args:
-            - "cp /default_prometheus_config/prometheus.yml /prometheus_configs/prometheus.yml && bin/yugaware -Dconfig.file=/data/application.docker.conf"
```
So we don't need [this](https://github.com/yugabyte/charts/pull/63/files#diff-58c816a5badea1634d1f7fd6bc835152552ed86e143de953e41b0953a9313deeR197-R198) anymore, right ?

baba230896 replied:
I guess this is something I missed while rebasing to remove the merge conflicts. Indeed, I'll remove the `volumeMount` related to default Prometheus conf in YW STS Pod Spec.

baba230896 committed:
Resolved review comments

1. Directly used cp in initContainer command at the place of sh.
2. Removed the volumeMount of default prometheus conf from YW STS Pod
Spec.

baba230896 commented:
@skorobogatydmitry Could you please review it again?

skorobogatydmitry commented:
> @skorobogatydmitry Could you please review it again?

Sure!
There is "request review" button in `Reviewers` section - this works better to let reviewer know to re-visit PR.

skorobogatydmitry reviewed (approved):
LGTM! Please collect one more approval.

bhavin192 reviewed (approved):
This looks good to me!
